Formula/vim.rb

```diff
@@ -16,6 +16,7 @@ class Vim < Formula
   option "override-system-vi", "Override system vi"
   option "disable-nls", "Build vim without National Language Support (translated messages, keymaps)"
   option "with-client-server", "Enable client/server mode"
+  option "enable-termtruecolor", "Include support for 24-bit colors for compatible terminals"
```

Comment:
@DomT4, @UniqMartin, According to [this](https://github.com/vim/vim/commit/8a633e3427b47286869aa4b96f2bfc1fe65b25cd) commit, it indicates that it is compatible with ISO-8613-3 terminals. 

I have not seen any issues with incompatible terminals. I tested this on iTerm2 (which does support 24-bit colors) and Terminal.app (which doesn't), and on the Linux side too with xfce4-terminal, (which also doesn't support it). So I agree that it can be the default because I haven't seen any issues from this.

But maybe we should still give users the option to disable it if they would like?